LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes

marta- opened the pull request:
This PR only makes the look of the Forms and Subjects pages consistent with their corresponding dashboard boxes. The form filters are not yet transfered from Dashboard to the Forms page when clicking on "Expand".

marta- committed:
LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes
Work in progress. Known issue: Displaying forms for a specific questionnaire ( ...content.html/Forms?questionnaire=<..>) no longer seems to work.

OrdiNeu commented on modules/data-entry/src/main/frontend/src/dataHomepage/Forms.jsx:
```diff
@@ -102,41 +72,23 @@ function Forms(props) {
-          />
-      </CardContent>
-    </Card>
+          questionnaire={questionnaireID}
```
Source of the individual-questionnaire-view bug
```suggestion
          questionnaire={questionnaireID[1]}
```

sashaandjic commented:
`Filters should be transferred from the origin dashboard widget (only when clicking on "Expand" in that box, not when clicking on the side menu entry).`
By "transfered" did you mean move the filter values?
The way it work now, the filters are shown on the Form page regardless of the way opening it and the Filter values never transfer.

OrdiNeu commented on modules/data-entry/src/main/frontend/src/dataHomepage/FormView.jsx:
```diff
@@ -38,11 +38,16 @@ import { Link } from 'react-router-dom';
   const [ activeTab, setActiveTab ] = useState(0);
+  const [ title, setTitle ] = useState(typeof(props.title) != 'undefined' ? props.title : "Forms");
+  const [ subtitle, setSubtitle ] = useState();
+  const [ questionnairePath, SetQuestionnairePath ] = useState();
```
```suggestion
  const [ questionnairePath, setQuestionnairePath ] = useState();
```
To match the call on line 91

OrdiNeu commented on modules/data-entry/src/main/frontend/src/dataHomepage/FormView.jsx:
```diff
@@ -68,12 +73,40 @@ function FormView(props) {
+        if (qData) {
+          setTitle(qData["title"]);
+          setSubtitle(qData["description"]);
+          setQuesionnairePath(qData["@path"]);
```
```suggestion
          setQuestionnairePath(qData["@path"]);
```
To match the above

sashaandjic commented:
The smaller row spacing on dashboard grids look so much better. The look of Subject and Forms pages would benefit from the same change. 
The app name and version at the bottom of the side bar look a bit like an error. Would orange or turquoise colured fonts improve the look?
I believe we can turn off the paginationTop on the Forms page as well.

OrdiNeu commented on modules/data-entry/src/main/frontend/src/dataHomepage/FormView.jsx:
```diff
@@ -68,12 +73,40 @@ function FormView(props) {
+
   return (
     <Card className={classes.formView}>
+      {(!props.expanded || !props.disableHeader && !props.disableAvatar && title) &&
```
Prefer these props to be pulled out in the `{...} = props;` line on 45, since it makes it more explicit that they exist.

marta- committed:
LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes
Fixed a bunch of embarassing typos

marta- committed:
LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes
Addressing review comments:
* Improved code style
* Pemoved top pagination for consisency

marta- committed:
LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes
Addressing review comments: Made boxes on Subjects and Forms pages look identical to the ones on the dashboard (only wider and with more columns)

marta- commented:
> * The smaller row spacing on dashboard grids look so much better. The look of Subject and Forms pages would benefit from the same change.
> * I believe we can turn off the paginationTop on the Forms page as well.

Done

> * The app name and version at the bottom of the side bar look a bit like an error. Would yellow or turquoise colored fonts improve the look? Or a line above it to separate it from the options?

To be done as a separate task/PR

marta- committed:
LFS-780: Forms and Subjects views should be consistent with the corresponding Dashboard boxes
Addressing review comments: more codestyle

marta- commented:
> `Filters should be transferred from the origin dashboard widget (only when clicking on "Expand" in that box, not when clicking on the side menu entry).`
> By "transfered" did you mean move the filter values?
> The way it work now, the filters are shown on the Form page regardless of the way opening it and the Filter values never transfer.

Yes, values should be transferred. This is not implemented in this PR.

sashaandjic reviewed (commented):
a very nice and polished look - tested - approved
